interpreter/interpretable.go

```diff
@@ -533,6 +533,18 @@ type evalVarArgs struct {
 	impl     functions.FunctionOp
 }
 
+// NewCall creates a new call Interpretable.
+func NewCall(id int64, function, overload string, args []Interpretable, trait int, impl functions.FunctionOp) InterpretableCall {
```

Comment:
I'm debating whether to remove 'trait' from the argument set since that should only be for receiver functions and decorators that modify receiver functions seem error-prone. Thoughts?

Reply:
I wanted to ditch it too, I just wasn't able to 100% convince myself it was okay. If you don't see any issues I'd love to remove it.

Reply:
The alternative would be to provide a more use case specific function like `OverrideCallImpl(InterpretableCall, FunctionOp)`

Reply:
I think just removing `trait` is fine because ideally the functions provided from a decorator aren't relying on an intrinsic property of the first argument.